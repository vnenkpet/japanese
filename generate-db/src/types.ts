const TYPES = {
  ExtractService: Symbol.for('ExtractService'),
  Main: Symbol.for('Main'),
};

export { TYPES };
