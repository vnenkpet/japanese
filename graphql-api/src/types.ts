export const Types = {
  DICTIONARY_ENTRY_REPOSITORY: Symbol.for('DICTIONARY_ENTRY_REPOSITORY'),
  DATABASE_CONNECTION: Symbol.for('DATABASE_CONNECTION'),
  AUTH_GUARD: Symbol.for('AUTH_GUARD'),
  CONFIG: Symbol.for('CONFIG'),
};
