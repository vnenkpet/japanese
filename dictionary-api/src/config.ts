import * as dotenv from 'dotenv';
import { inject, toNumber, toString } from 'typescript-stringcaster';
dotenv.config();

const source = process.env;

class Config {
  @inject({ cast: toString, source })
  public MONGODB_URI: string;

  @inject({ cast: toString, source })
  public MONGODB_URI_TEST: string;

  @inject({ cast: toNumber, source })
  public PORT: string;

  @inject({ cast: toString, source, defaultValue: 'development' })
  public NODE_ENV: string;
}

export default new Config();
