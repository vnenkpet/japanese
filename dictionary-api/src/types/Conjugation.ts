import { Field, ObjectType } from "type-graphql";

@ObjectType()
export default class Conjugation {
  @Field() public name: string;

  @Field() public form: string;
}
