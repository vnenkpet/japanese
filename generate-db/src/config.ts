import { injectable } from 'inversify';
import { IConfig } from './IConfig';

@injectable()
export class Config implements IConfig {
  get jmdictArchiveUrl(): string {
    return (
      process.env.JMDICT_ARCHIVE_URL ||
      'https://github.com/scriptin/jmdict-simplified/releases/download/3.0.1/jmdict-eng-3.0.1.json.tgz'
    );
  }

  get jmnedictArchiveUrl(): string {
    return (
      process.env.JMNEDICT_ARCHIVE_URL ||
      'https://github.com/scriptin/jmdict-simplified/releases/download/3.0.1/jmnedict-3.0.1.json.tgz'
    );
  }

  get mongoConnectionUri(): string {
    return (
      process.env.MONGO_CONNECTION_URI || 'mongodb://localhost:27017/jmdict'
    );
  }
}
