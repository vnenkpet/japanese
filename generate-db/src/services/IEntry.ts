export interface IEntry {
  id: string;
  kanji: [
    {
      common: false;
      text: string;
      tags: [];
    }
  ];
  kana: [
    {
      common: true;
      text: string;
      tags: [];
      appliesToKanji: [string];
    }
  ];
  sense: [
    {
      partOfSpeech: [string];
      appliesToKanji: [string];
      appliesToKana: [string];
      related: [string];
      antonym: [string];
      field: [string];
      dialect: [string];
      misc: [string];
      info: [string];
      languageSource: [string];
      gloss: [
        {
          lang: string;
          text: string;
        }
      ];
    }
  ];
}
