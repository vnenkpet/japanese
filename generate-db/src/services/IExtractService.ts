export type DataProcessFunction = (data: any) => void;

export interface IExtractService {
  /**
   * Extracts a tar.gz file in-memory, parses the JSON from it using a stream and processes data using the provided function
   *
   * @param readStream
   * @param state
   * @param processData
   * @param callback
   */
  extractJsonFromStream: (
    readStream: NodeJS.ReadableStream,
    path: string,
    processData: DataProcessFunction,
  ) => Promise<{}>;
}
