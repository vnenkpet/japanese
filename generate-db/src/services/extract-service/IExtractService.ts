export type DataProcessFunction = (data: any) => Promise<void>;

export interface IExtractService {
  /**
   * Extracts a tar.gz file in-memory, parses the JSON from it using a stream and processes data using the provided function
   *
   * @param readStream Read stream from the .tar.gz file where the json file is stored
   * @param path For example "items.*" to parse only array items in a json document looking like `{name: "Example file", items: [{id: 1, id: 2}]}`
   * @param processData Processing function run on each item extracted from the `path` in the json file
   */
  extractJsonFromStream: (
    readStream: NodeJS.ReadableStream,
    path: string,
    processData: DataProcessFunction,
  ) => Promise<{}>;
}
