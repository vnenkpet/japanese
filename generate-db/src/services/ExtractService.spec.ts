import 'reflect-metadata';

import { container } from '../inversify.config';
import { TYPES } from '../types';
import { IExtractService } from './IExtractService';
import * as fs from 'fs';
import { IEntry } from './IEntry';

describe('Export service', () => {
  const extractService = container.get<IExtractService>(TYPES.ExtractService);

  it('Service is defined', async () => {
    expect(extractService).toBeDefined();
  });

  it('Read the size of the file', async () => {
    const str = fs
      .readFileSync('./src/services/__data__/jmdict-eng-common-3.0.1.json')
      .toString('utf8');
    const parsed = JSON.parse(str);
    expect(parsed.words.length).toMatchSnapshot();
  });

  it('Succesfully extracts example archive', async () => {
    const entries: IEntry[] = [];

    await extractService.extractJsonFromStream(
      fs.createReadStream(
        './src/services/__data__/exampleJsonContent.json.tgz',
      ),
      'words.*',
      data => {
        if (data.id) {
          entries.push(data);
        }
      },
    );

    expect(entries.length).toBe(3);
    expect(entries[0]).toMatchObject({
      id: '1000110',
      kanji: [
        {
          common: true,
          text: 'ＣＤプレーヤー',
          tags: [],
        },
        {
          common: false,
          text: 'ＣＤプレイヤー',
          tags: [],
        },
      ],
      kana: [
        {
          common: true,
          text: 'シーディープレーヤー',
          tags: [],
          appliesToKanji: ['ＣＤプレーヤー'],
        },
        {
          common: false,
          text: 'シーディープレイヤー',
          tags: [],
          appliesToKanji: ['ＣＤプレイヤー'],
        },
      ],
      sense: [
        {
          partOfSpeech: ['n'],
          appliesToKanji: ['*'],
          appliesToKana: ['*'],
          related: [],
          antonym: [],
          field: [],
          dialect: [],
          misc: [],
          info: [],
          languageSource: [],
          gloss: [
            {
              lang: 'eng',
              text: 'CD player',
            },
          ],
        },
      ],
    });
  });
});
