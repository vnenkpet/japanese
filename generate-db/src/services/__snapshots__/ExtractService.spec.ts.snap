// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Export service Read the size of the file 1`] = `22350`;
